src/openforms/emails/digest.py

```diff
@@ -69,6 +77,76 @@ class BrokenConfiguration:
     exception_message: str
 
 
+@dataclass
+class InvalidRegistrationBackend:
+    config_name: StrOrPromise
+    exception_message: str
+    form_name: str
+    form_id: str = ""
+    form_level: bool = False
+
+    @property
+    def admin_link(self) -> str:
+        form_admin_url = reverse(
+            "admin:forms_form_change", kwargs={"object_id": self.form_id}
+        )
+        return form_admin_url
+
+
+@dataclass
+class InvalidLogicRule:
+    variable: str
+    form_name: str
+    form_id: str
+
+    @property
+    def admin_link(self) -> str:
+        return reverse("admin:forms_form_change", kwargs={"object_id": self.form_id})
+
+
+def is_form_config_valid(plugin: BasePlugin, options: dict[str, str]) -> bool:
+    serializer = plugin.configuration_options(data=options)
+
+    try:
+        serializer.is_valid(raise_exception=True)
+    except serializers.ValidationError:
+        return False
+
+    return True
+
+
+def check_general_config(plugin: BasePlugin) -> str:
+    try:
+        plugin.check_config()
+
+    # we always raise an InvalidPluginConfiguration exception even when we have
+    # errors like HTTPError
+    except InvalidPluginConfiguration as e:
+        return e.args[0]
+
+    return ""
+
+
+def get_variables_from_logic(form_logic: FormLogic) -> list[str]:
```

Comment:
Also, instead of this whole new implementation, it's probably better to use the existing introspection utilities: `openforms.utils.json_logic.introspect_json_logic` which returns a `ExpressionIntrospection` instance that has the method `get_input_components` that can serve as inspiration or even be adapted (it seems to not be used anymore).